pkg/server/httpAPI.go

```diff
@@ -432,6 +437,71 @@ func (h *HTTPHandler) serveConfig(rw http.ResponseWriter, req *http.Request) {
 	domaiName = fmt.Sprintf("%s.%s.%s", svcName, namespace, "svc")
 	config["advertise-client-urls"] = fmt.Sprintf("%s://%s.%s:%s", protocol, podName, domaiName, clientPort)
 
+	//Read sts spec for updated replicas to toggle `initial-cluster-state`
+	clientSet, err := miscellaneous.GetKubernetesClientSetOrError()
+	if err != nil {
+		h.Logger.Errorf("failed to create clientset: %v", err)
+		return
+	}
+	curSts := &appsv1.StatefulSet{}
+	errSts := clientSet.Get(context.TODO(), client.ObjectKey{
+		Namespace: podNS,
+		Name:      podName[:strings.LastIndex(podName, "-")],
+	}, curSts)
+	if errSts != nil {
+		h.Logger.Warn("error fetching etcd sts ", errSts)
+		return
+	}
+
+	config["initial-cluster-state"] = "new"
+	//TODO: achieve this without an sts?
+	if *curSts.Spec.Replicas > 1 && *curSts.Spec.Replicas > curSts.Status.UpdatedReplicas {
+		config["initial-cluster-state"] = "existing"
+	}
+	// TODO Remove below prints
+	fmt.Println("*curSts.Spec.Replicas is ", *curSts.Spec.Replicas)
+	fmt.Println("curSts.Status.UpdatedReplicas is ", curSts.Status.UpdatedReplicas)
+
+	//INITIAL_CLUSTER served via the etcd config must be tailored to the number of members in the cluster at that point. Else etcd complains with error "member count is unequal"
+	//One reason why we might want to have a strict ordering when members are joining the cluster
+	//addmember subcommand achieves this by making sure the pod with the previous index is running before attempting to add itself as a learner
+	etcdConn := *h.EtcdConnectionConfig
+	etcdConn.Endpoints = []string{fmt.Sprintf("%s://%s:%s", protocol, domaiName, peerPort)} //[]string{"http://etcd-main-peer.default.svc.cluster.local:2380"} //TODO: pass via env var
+	clientFactory := etcdutil.NewFactory(etcdConn)
+	cli, err := clientFactory.NewCluster()
+	if err != nil {
+		h.Logger.Warnf("Error with NewCluster() : %v", err)
+	}
+	ctx, cancel := context.WithTimeout(context.TODO(), brtypes.DefaultEtcdConnectionTimeout)
+	defer cancel()
+	memList, err := cli.MemberList(ctx)
+	noOfMembers := 0
+	if err != nil {
+		h.Logger.Warnf("Error with MemberList() : %v", err)
```

Comment:
If there is an error then, do you assumed that there will be 0 members? Is that a correct assumption?

Reply:
Yes, if there's an error here, the assumption is that it is a member bootstrap or a cluster is waking up from hibernation, so we would not want to manipulate the `initial-cluster` string